contrib/socat/patches/fix-test.patch

```diff
@@ -0,0 +1,99 @@
+--- a/Makefile.in
++++ b/Makefile.in
+@@ -190,7 +190,7 @@
+ 
+ # perform some tests on socat
+ test: progs
+-	./test.sh
++	./test.sh --expect-fail 478,528
+ 
+ cert:
+ 	# prepare critical files with correct permissions to avoid race cond
+--- a/test.sh
++++ b/test.sh
+@@ -8,12 +8,12 @@
+ # this script uses functions; you need a shell that supports them
+ 
+ # you can pass general options to socat: export OPTS="-d -d -d -d -lu"
+-# you can eg strace socat with: export TRACE="strace -v -tt -ff -D -x -s 1024 -o /tmp/$USER/socat.strace"
```

Comment:
rather than patch 50 uses of USER you could set it in the environment

Reply:
It was done like that because test 217 was failing whenever $USER was set in the environment.

Since then, i've discovered tons of heisenbugs in the testing code.  
So I've reverted back to disabling tests.